v1/lib/core/utils.js

```diff
@@ -38,53 +38,90 @@ function idx(target, keyPaths) {
   );
 }
 
-function getGitLastUpdated(filepath) {
-  function isTimestamp(str) {
-    return /^\d+$/.test(str);
+const GIT_LAST_UPDATED_TYPE = {
+  TIME: 'time',
+  AUTHOR: 'author',
+};
+
+function getGitLastUpdated(filepath, type) {
+  if (Object.values(GIT_LAST_UPDATED_TYPE).indexOf(type) === -1) {
+    return null;
   }
 
   // Wrap in try/catch in case the shell commands fail (e.g. project doesn't use Git, etc).
   try {
+    const format = GIT_LAST_UPDATED_TYPE.TIME === type ? '%ct' : 'author=%an';
     // To differentiate between content change and file renaming / moving, use --summary
     // To follow the file history until before it is moved (when we create new version), use
     // --follow.
     const silentState = shell.config.silent; // Save old silent state.
     shell.config.silent = true;
     const result = shell
-      .exec(`git log --follow --summary --format=%ct ${filepath}`)
+      .exec(`git log --follow --summary --format=${format} ${filepath}`)
       .stdout.trim();
     shell.config.silent = silentState;
 
-    // Format the log results to be
+    // Format the log results to be either
     // ['1234567', 'rename ...', '1234566', 'move ...', '1234565', '1234564']
+    // OR
+    // ['Yangshun Tay', 'rename ...', 'Endiliey', 'move ...', 'Joel', 'Fienny']
     const records = result
       .toString('utf-8')
       .replace(/\n\s*\n/g, '\n')
       .split('\n')
       .filter(String);
-
-    const timeSpan = records.find((item, index, arr) => {
-      const currentItemIsTimestamp = isTimestamp(item);
-      const isLastTwoItem = index + 2 >= arr.length;
-      const nextItemIsTimestamp = isTimestamp(arr[index + 1]);
-      return currentItemIsTimestamp && (isLastTwoItem || nextItemIsTimestamp);
-    });
-
-    if (timeSpan) {
-      const date = new Date(parseInt(timeSpan, 10) * 1000);
-      return date.toLocaleString();
-    }
+    return records;
   } catch (error) {
     console.error(error);
   }
+  return null;
+}
+function getGitLastUpdatedTime(filepath) {
+  function isTimestamp(str) {
+    return /^\d+$/.test(str);
+  }
+
+  const records = getGitLastUpdated(filepath, GIT_LAST_UPDATED_TYPE.TIME);
+
+  const timeSpan = records.find((item, index, arr) => {
```

Comment:
This finding logic can be shifted into the `getGitLastUpdated` method.

And `getGitLastUpdatedTime` will in the end just become:

```js
function getGitLastUpdatedTime(filepath) {
  const commit = getGitLastUpdated(filepath);
  if (!commit) {
    return null;
  }

  const date = new Date(parseInt(commit.timestamp, 10) * 1000);
  return date.toLocaleString();
}
```